parsons/mobilecommons/mobilecommons.py

```diff
@@ -0,0 +1,427 @@
+                    response_dict["response"][first_data_key][second_data_key]
+                )
+                # Append to final table
+                final_table.concat(response_table)
+                final_table.materialize()
+
+        return Table(final_table[:limit])
+
+    def _check_response_status_code(self, response):
+        """
+        A helper function that checks the status code of a response and raises an error if the
+        response code is not 200
+
+        `Args:`
+            response: requests package response object
+        """
+        if response.status_code != 200:
+            error = f"Response Code {str(response.status_code)}"
+            error_html = BeautifulSoup(response.text, features="html.parser")
+            error += "\n" + error_html.h4.next
+            error += "\n" + error_html.p.next
+            raise HTTPError(error)
+
+    def _parse_get_request(self, endpoint, params):
+        """
+        A helper function that sends a get request to MobileCommons and then parses XML responses in
+        order to return the response as a dictionary
+
+        `Args:`
+            endpoint: str
+                The endpoint, which will be appended to the base URL for each request
+            params: dict
+                Parameters to be passed into GET request
+        `Returns:`
+            xml response parsed into list or dictionary
+        """
+        response = self.client.request(endpoint, "GET", params=params)
+
+        # If there's an error with initial response, raise error
+        self._check_response_status_code(response)
+
+        # If good response, compile data into final_table
+        # Parse xml to nested dictionary and load to parsons table
+        response_dict = xmltodict.parse(
+            response.text, attr_prefix="", cdata_key="", dict_constructor=dict
+        )
+        return response_dict
+
+    def _mc_post_request(self, endpoint, params):
+        """
+        A function for POST requests that handles MobileCommons xml responses
+
+        `Args:`
+            endpoint: str
+                The endpoint, which will be appended to the base URL for each request
+            params: dict
+                Parameters to be passed into GET request
+        `Returns:`
+            xml response parsed into list or dictionary
+        """
+
+        response = self.client.request(endpoint, "POST", params=params)
+
+        response_dict = xmltodict.parse(
+            response.text, attr_prefix="", cdata_key="", dict_constructor=dict
+        )
+        if response_dict["response"]["success"] == "true":
+            return response_dict["response"]
+        else:
+            raise HTTPError(response_dict["response"]["error"])
+
+    def get_broadcasts(
+        self, first_date=None, last_date=None, status=None, campaign_id=None, limit=None
+    ):
+        """
+        A function for get broadcasts
+
+        `Args:`
+            first_date: str
```

Comment:
You marked this as resolved, but I don't see the additional detail here that I was expecting. So if I saw this documentation, I'd wonder "what is the string format for first_date? can I use `January 1st, 2023` or `1-1-23` or `01-01-23` or what?" They can't even figure out the format form looking at the MobileCommons docs, because you're actually doing some intervening work with the format_helper method.

Reply:
Ugh, nevermind - Github was showing me old versions of your files. Ignore this comment!